articles/openshift/howto-deploy-java-eap-app.md

```diff
@@ -0,0 +1,384 @@
+
+1. Open `http://localhost:8080/` in your browser to visit the application home page. If you have created more instances, you can access them by shifting the port number, for example `http://localhost:9080/`. The application will look similar to the following image:
+
+    ![ToDo EAP demo Application](./media/howto-deploy-java-eap-app/todo-demo-application.png)
+
+1. Check the application health endpoints. These endpoints will be used by OpenShift to verify when your pod is live and ready to receive user requests:
+
+   ```bash  
+    todo-list (bootable-jar) $ curl http://localhost:9990/health/live
+    {"status":"UP","checks":[{"name":"SuccessfulCheck","status":"UP"}]}
+    todo-list (bootable-jar) $ curl http://localhost:9990/health/ready
+    {"status":"UP","checks":[{"name":"deployments-status","status":"UP","data":{"todo-list.war":"OK"}},{"name":"server-state","status":"UP","data":{"value":"running"}},{"name":"boot-errors","status":"UP"},{"name":"DBConnectionHealthCheck","status":"UP"}]}
+    ```
+
+1. Press **Control-C** to stop the application.
+1. Execute the following command to stop the database server:
+
+    ```bash
+    docker stop mssqlserver
+    ```
+
+1. If you don't plan to use the docker database again, execute the following command to remove the database server from your docker registry:
+
+    ```bash
+    docker rm mssqlserver
+    ```
+
+## Deploy to OpenShift
+
+Before deploying the demo application on OpenShift we will deploy the database server. The database server will be deployed by using a [DeploymentConfig OpenShift API resource](https://docs.openshift.com/container-platform/4.8/applications/deployments/what-deployments-are.html#deployments-and-deploymentconfigs_what-deployments-are). The database server deployment configuration is available as a YALM file on the application source code. To deploy the application, we are going to use the JBoss EAP Helm Charts that are already available in ARO. We also need to supply the desired configuration, for example, the database user, the database password, the driver version we want to use, and the connection information used by the data source. Since this information contains sensible information, we will use OpenShift secrets objects to store it.
+
+> [!NOTE]
+> You can also use the JBoss EAP Operator to deploy this example, however, notice that the JBoss EAP Operator will deploy the application as stateful sets. Use the JBoss EAP Operator if your application requires one or more one of the following.
```

Comment:
It could be implied by the below list but Transactions are a notable reason to use the Operator

Reply:
+1

People don't understand whether they need stable, persistent storage and likely think they don't.

Reply:
I don't have the ability to mark this as resolved but this area could be marked as resolved for me

Reply:
I am happy with however you prefer to work with this review. What I suppose I am saying is that you have addressed my initial feedback as Transaction recovery is now listed below